### ISSUE TITLE
scan extensions (not working)

### ISSUE THREAD

**Issue by amster699** [ociredefz/vsaudit#1]
hello guys i like alot your tool but im having some isues about scaning extensions bicouse wend i type (en 100-1000) its never grab any extensions and i also try with sipvicious and it grabs extension Nr 500 and with vsaudit never grabs any extension... how can i fix this ?
thanks and regards

**Comment by eurialo** [comment]
Hi, is there the `REGISTER` method enabled on the system target?
Actually all vsaudit requests are based on `OPTIONS/REGISTER` methods for enumeration/bruteforce actions (I think that sipvicious have multi request type check).

You can check if method is available with: `info [target-address]` that shows you for example:  
`+ Allows: INVITE, ACK, CANCEL, OPTIONS, BYE, REFER, SUBSCRIBE, NOTIFY, INFO, ..
`

Sorry but I have no free time to develop the new features, I will do as soon as possible!

**Comment by ociredefz** [comment]
Hi @amster699!

Actually `vsaudit` only supports the `REGISTER` method in both process (`enumeration/bruteforce`), the "extension" will be found (and listed) if It's configured in `Asterisk` with no password or if Its password combination will be found (evermore in both process). 

This limit exists because the answer to the `REGISTER` request for any `extension` name (we're trying to check) is the same (`401 Unauthorized` as I remember). 

I think I'll setup new features as soon as possible using others method than `REGISTER`.
Any suggestions/contributions are welcome!

**Comment by amster699** [comment]
yes brother wend i use sipvicious i usually use comands like this and after
i test the same on Vsaudit to check it up and never grabs any extension on
vsaudit
n1: svmap.py (target ip)/24 (check extensions)
n2: svwar.py -D -m OPTIONS (target ip)
after i guet the extensions (noauth) i usualy go to vsaudit and try to
bruteforce the extension and never working but now i know why that happend
and i also see sometimes changes must be made to test it on WAN i tough
mayby i had to open ports and allow my firewall setings  haha but im really
thanks for your help brother
please let me know wend you make any update on VSaudit bro i really like
your tool
be safe mo brother thanks

2017-06-10 17:04 GMT+01:00 Federico Fazzi <notifications@github.com>:
